internal/repository/repository.go

```diff
@@ -0,0 +1,45 @@
+package repository
+
+import (
+	"github.com/benchttp/server/internal"
+	"github.com/benchttp/server/internal/store"
+	"github.com/google/uuid"
+)
+
+// Repository exposes the available operations to access the data layer.
+type Repository struct {
+	store store.Store
+}
+
+// New returns a new repository.
+// For now it uses store.NewDefault as the internal store.
+func New() (*Repository, error) {
+	return &Repository{
+		store: store.NewDefault(),
+	}, nil
+}
+
+// StoreReport stores a report in the data layer
+// and returns the key to retrieve it later on.
+func (r *Repository) StoreReport(data internal.Report) (string, error) {
+	id, err := uuid.NewRandom()
```

Comment:
[question] For some reason I initially thought we'd need to create the uuid from the runner but I actually see no reason, any thoughts?

Reply:
Yeah no reason as it is not needed. The runner would received the id of the run it just send from the api response and print it for the user.
It's even safer like that isn't it?